api/src/nv_ingest_api/util/dataloader/dataloader.py

```diff
@@ -0,0 +1,177 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright (c) 2025, NVIDIA CORPORATION.
+import fsspec
+from upath import UPath as Path
+from abc import ABC, abstractmethod
+import queue
+import threading
+import subprocess
+import json
+import logging
+import math
+import importlib
+
+logger = logging.getLogger(__name__)
+
+if importlib.util.find_spec("ffmpeg") is None:
+    logger.error(
+        "Unable to load the Dataloader,ffmpeg was not installed, "
+        "please install it using `pip install ffmpeg-python` and `apt-get install ffmpeg`"
+    )
+else:
+    import ffmpeg
+
+
+class LoaderInterface(ABC):
+
+    @abstractmethod
+    def split(self, input_path: str, output_dir: str, split_interval: int = 0):
+        pass
+
+
+def _probe(filename="pipe:", cmd="ffprobe", format=None, file_handle=None, timeout=None, **kwargs):
+    args = [cmd, "-show_format", "-show_streams", "-of", "json"]
+    args += ffmpeg._utils.convert_kwargs_to_cmd_line_args(kwargs)
+    args += ["pipe:"]
+
+    p = subprocess.Popen(args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    communicate_kwargs = {}
+    if timeout is not None:
+        communicate_kwargs["timeout"] = timeout
+    if file_handle:
+        communicate_kwargs["input"] = file_handle
+    out, err = p.communicate(**communicate_kwargs)
+    if p.returncode != 0:
+        raise ffmpeg._run.Error("ffprobe", out, err)
+    return json.loads(out.decode("utf-8"))
+
+
+class MediaInterface(LoaderInterface):
```

Comment:
One thing that would be nice here would be to be able to seek/split based on bytes, time, or frame offset; it makes chopping up the file more intutitive.

This would make it easier for end users who want to do something like splitting into 60 second chunks, or who want to process based on specific offsets in a file or based on other meta information.